onnxruntime/core/providers/openvino/backend_manager.cc

```diff
@@ -125,6 +128,24 @@ BackendManager::BackendManager(const GlobalContext& global_context,
         } catch (std::string const& msg) {
           ORT_THROW(msg);
         }
+      } else if (device_type.find("NPU") != std::string::npos &&
```

Comment:
MSFT uses build with "OPENVINO_DISABLE_NPU_FALLBACK"directive. 
From the code flow it looks like only when the provider option is enabled or if either of the conditions fail , 
```
if (device_type.find("NPU") != std::string::npos &&
          !GetGlobalContext().disable_cpu_fallback &&
          !ep_ctx_handle_.IsValidOVEPCtxGraph())
```
the exception is thrown. 

Can you throw the exception generic ?

Reply:
Updated